cmake/compiler/xt-clang/target.cmake

```diff
@@ -1,3 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 include(${ZEPHYR_BASE}/cmake/compiler/xcc/target.cmake)
+
+if("${ARCH}" STREQUAL "xtensa")
```

Comment:
`xt-clang` is clang for xtensa, so why this test and CMake file split, as all this CMake code is for the `xtensa` architecture ?

Reply:
@tejlmand could you re-check, please? I addressed your comment last Friday, 5 days ago.